internal/node/indexer/server.go

```diff
@@ -48,21 +49,30 @@ type Server struct {
 
 func (s *Server) Run(ctx context.Context) error {
 	var (
-		// TODO Develop a more effective solution to implement back pressure.
-		tasksChan = make(chan *engine.Tasks)
+		// TODO Develop a more effective solution to implement back pressure. - Resolving using utils.TaskBuffer
 		errorChan = make(chan error)
 	)
 
 	zap.L().Info("start node", zap.String("version", constant.BuildVersion()))
 
-	s.source.Start(ctx, tasksChan, errorChan)
+	s.source.Start(ctx, s.TaskBuffer, errorChan)
 
 	for {
 		select {
-		case tasks := <-tasksChan:
+		case err := <-errorChan:
+			if err != nil {
+				return fmt.Errorf("an error occurred in the source: %w", err)
+			}
+
+			return nil
+
+		default:
+			// Get task from the task buffer
+			task := s.TaskBuffer.Get()
+
 			retryableFunc := func() error {
-				if err := s.handleTasks(ctx, tasks); err != nil {
-					return fmt.Errorf("handle tasks: %w", err)
+				if err := s.handleTasks(ctx, task); err != nil {
+					errorChan <- fmt.Errorf("handle tasks error: %w", err)
```

Comment:
```suggestion
I think the function retryableFunc should return the error instead of sending it to the errorChan. This approach can make the retry mechanism clearer and more straightforward
```

Reply:
Resolved.